packages/@aws-cdk/core/lib/environment-component.ts

```diff
@@ -0,0 +1,130 @@
+  readonly region: EnvComponent;
+}
+
+/**
+ * An enum-like class that represents the result of comparing either two {@link EnvComponent}s,
+ * or a {@link EnvComponent}, and a string.
+ */
+export class EnvComponentComparison {
+  /**
+   * This means we're certain the two components are NOT
+   * Tokens, and identical.
+   */
+  public static readonly SAME = new EnvComponentComparison();
+
+  /**
+   * This means we're certain the two components are NOT
+   * Tokens, and different.
+   */
+  public static readonly DIFFERENT = new EnvComponentComparison();
+
+  /** This means exactly one of the components is a Token. */
+  public static readonly ONE_UNRESOLVED = new EnvComponentComparison();
+
+  /** This means both components are Tokens. */
+  public static readonly BOTH_UNRESOLVED = new EnvComponentComparison();
+
+  private constructor() {
+  }
+
+  /**
+   * Returns true if:
+   *   - Neither component was a Token, and they were equal to each other.
+   * Or:
+   *   - Both component were Tokens.
+   *
+   * @returns true if `this` is either SAME or BOTH_UNRESOLVED,
+   *   false if `this` is either DIFFERENT or ONE_UNRESOLVED
+   */
+  // public equalOrBothUnresolved(): boolean {
+  //   return this === EnvComponentComparison.SAME ||
+  //     this === EnvComponentComparison.BOTH_UNRESOLVED;
+  // }
+
+  /**
+   * Returns true if:
+   *   - Neither component was a Token, and they were equal to each other.
+   * Or:
+   *   - Either component (or both!) was a Token.
+   *
+   * @returns true if `this` is either SAME or ONE_UNRESOLVED or BOTH_UNRESOLVED,
+   *   false if `this` is DIFFERENT
+   */
+  // public equalOrAnyUnresolved(): boolean {
+  //   return this !== EnvComponentComparison.DIFFERENT;
+  // }
+
+  /**
+   * Returns true if:
+   *   - Neither component was a Token, and they were not equal to each other.
+   * Or:
+   *   - Exactly one of the components was a Token.
+   *
+   * @returns true if `this` is either DIFFERENT or ONE_UNRESOLVED,
+   *   false if `this` is either SAME or BOTH_UNRESOLVED
+   */
+  // public unEqual(): boolean {
+  //   return !this.equalOrBothUnresolved();
+  // }
+}
+
+/**
+ * Represents a component of an Environment,
+ * like an AWS account ID,
+ * or the code name of an AWS region.
+ * Since they can be Tokens representing deploy-time values like AWS::AccountId and AWS::Region,
+ * their comparison logic is slightly more complicated than just comparing strings for equality.
+ */
+export class EnvComponent {
+  private readonly component: string;
+
+  public constructor(component: string) {
+    this.component = component;
+  }
+
+  /**
+   * Compare two components with each other.
+   */
+  public compare(envComponent: EnvComponent): EnvComponentComparison {
+    return this.compareToString(envComponent.toString());
+  }
+
+  /**
+   * Compare this component to a string representation.
+   */
+  public compareToString(thatComponent: string): EnvComponentComparison {
```

Comment:
This feels like it belongs to the tokens world. `Token.compare()` or something.

Reply:
Changed to `Token.compareStrings()`.